### PR TITLE
Refactor PyTorch JNI build

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Review by roywei — APPROVED** [review]

File: pytorch/pytorch-native/build.gradle [hunk 1]
```diff
@@ -76,22 +77,25 @@ task releaseJNI {
...
-            downloadBuild("win", "cpu")
-            cleanBuild()
-            downloadBuild("win", "cu101")
-            cleanBuild()
-            downloadBuild("win", "cu102")
-            cleanBuild()
-            downloadBuild("win", "cu110")
-            cleanBuild()
-        } else if (System.properties['os.name'].toLowerCase(Locale.ROOT).contains("mac")) {
+            if (!isGPU) {
```
Comment (frankfliu) [inline pytorch/pytorch-native/build.gradle:80]: This is just optimization, right?

File: .github/workflows/native_jni_s3_pytorch.yml [hunk 2]
```diff
@@ -55,13 +59,9 @@ jobs:
...
-        run: ./gradlew :pytorch:pytorch-native:releaseJNI
-      - name: Upload compiled jni library
-        uses: actions/upload-artifact@v1
-        if: always()
-        with:
-          name: jnilib-${{ runner.os }}-GPU
-          path: pytorch/pytorch-native/jnilib
+        run: ./gradlew :pytorch:pytorch-native:releaseJNI -Pgpu
+      - name: Copy files to S3 with the AWS CLI
+        run: aws s3 sync pytorch/pytorch-native/jnilib s3://djl-ai/publish/pytorch-1.7.1/jnilib
```
Comment (frankfliu) [inline .github/workflows/native_jni_s3_pytorch.yml:64]: Let's invalidate the cache as well

File: .github/workflows/native_jni_s3_pytorch.yml [hunk 3]
```diff
@@ -12,7 +12,13 @@ jobs:
...
       matrix:
         operating-system: [macos-latest, ubuntu-16.04, windows-latest]
-
+        include:
+          - operating-system: macos-latest
+            os: osx
+          - operating-system: ubuntu-16.04
+            os: linux
+          - operating-system: windows-latest
+              os: win
```
Comment (frankfliu) [inline .github/workflows/native_jni_s3_pytorch.yml:21]:
```suggestion
            os: win
```

File: pytorch/pytorch-native/build.gradle (hunk 1 above)
Comment (stu1130) [reply pytorch/pytorch-native/build.gradle:80]: releaseJNI used to compile cpu and gpu. If we split upload jobs, there might be a chance where both task uploads to the same file to s3

**Review by frankfliu — APPROVED** [review]

**Pull request merged**